[16.0][IMP] stock_release_channel: Add a company setting to enable/disable recomputation of channel on release

QuocDuong1306 opened the pull request.

simahawk committed:
shopfloor_mobile: detail-card add slot after details

simahawk committed:
shopfloor_mobile: detail-picking add slot after detail

simahawk committed:
shopfloor_mobile: add speed-dial component

simahawk committed:
shopfloor_mobile: add picking edit form

simahawk committed:
shopfloor_mobile: screen change empty content text

simahawk committed:
shopfloor_mobile: fix web manifest

simahawk committed:
shopfloor_mobile: list add divider

simahawk committed:
shopfloor_mobile: make app background light grey and configurable

simahawk committed:
shopfloor_mobile: imp/fix all detail info screens

simahawk committed:
shopflor_mobile: demo.core.make_picking add type code

simahawk committed:
shopfloor_mobile: bump 13.0.1.3.0

simahawk committed:
shopfloor_mobile: allow edit picking for all types

Pickings are editable even for not outgoing type.
Most of the times operators change carrier during checkout.

simahawk committed:
shopfloor_mobile: bump 13.0.1.3.1

simahawk committed:
shopfloor_mobile: loc.cont.transfer fix label

simahawk committed:
shopfloor_mobile: get available lang from registry

simahawk committed:
shopfloor_mobile: split i18n lang files

simahawk committed:
shopfloor_mobile: add more translation keys

simahawk committed:
shopfloor_mobile: bump 13.0.1.4.0

oca-travis committed:
[UPD] Update shopfloor_mobile.pot

OCA-git-bot committed:
[ADD] icon.png

simahawk committed:
shopfloor_mobile: unify list-item loud title klass

Rename it from 'bold_title' to 'loud_title' as done for detail fields.

simahawk committed:
shopfloor_mobile: improve zone picking zone select

OCA-git-bot committed:
shopfloor_mobile 13.0.1.4.1

simahawk committed:
shopfloor_mobile: adapt demo.zone_picking data

simahawk committed:
shopfloor_mobile: enable Vue.config.devtools in demo mode

simahawk committed:
shopfloor_mobile: zone_picking add counters per zone

simahawk committed:
shopfloor_mobile: fix demo mode button

simahawk committed:
shopfloor_mobile: improve btn-back

simahawk committed:
shopfloor_mobile: zone_picking improve go back

Do not rely on history.back, instead reload the relevant screen
to always have fresh data.

simahawk committed:
shopfloor_mobile: zone_pickin unify UI for select op type

simahawk committed:
shopfloor_mobile: fix loading text

simahawk committed:
shopfloor_mobile: fix text color class usage

simahawk committed:
shopfloor_mobile: display menu item counters

simahawk committed:
shopfloor_mobile: upgrade vuetify.js v2.3.18

simahawk committed:
shopfloor_mobile: refresh menu when drawer expands

simahawk committed:
shopfloor_mobile bump 13.0.1.5.0

simahawk committed:
shopfloor_mobile: checkout disable no_package via options

simahawk committed:
shopfloor_mobile: show back btn on change pack

simahawk committed:
shopfloor_mobile: update ROADMAP

simahawk committed:
shopfloor_mobile: menu hide counters w/ no lines

OCA-git-bot committed:
shopfloor_mobile 13.0.1.6.0

simahawk committed:
shopfloor_mobile bump 13.0.1.6.1

OCA-git-bot committed:
[UPD] README.rst

sebastienbeau committed:
[IMP] black, isort, prettier

sebastienbeau committed:
[IMP] set module uninstallable

simahawk committed:
shopfloor_mobile: zone_picking wide screen fix widgets refresh

Compute a unique key for custom table widgets
otherwise when changing ordering by location/priority
the widgets are not updated (eg: wrong qty displayed)

simahawk committed:
shopfloor_mobile bump 13.0.1.6.2

Commit:
[FIX] shopfloor_mobile, delivery, rename button 'Mark as done' to 'Make partial delivery'

Because as soon as all lines are processed (qty done filled), the
related transfer is automatically validated. So the current button only
serves one purpose: creating a partial delivery.

OCA-git-bot committed:
shopfloor_mobile 13.0.1.6.3

simahawk committed:
shopfloor_mobile: odoo service allow custom headers

You can now pass custom headers for requests at init
or update them on an existing client instance before making the request.

simahawk committed:
shopfloor_mobile: zone picking rely on special headers

'zone_location' and 'picking_type' are now sent via headers.
This way they are shared for each call.

simahawk committed:
shopfloor_mobile: disable autocomplete on searchbar

simahawk committed:
shopfloor_mobile: display user info

simahawk committed:
shopfloor_mobile: checkout.select_package display vendor code

Commit:
Remove warehouse from profile

The warehouse of the picking type only should be used.

OCA-git-bot committed:
shopfloor_mobile 13.0.1.6.4

OCA-git-bot committed:
shopfloor_mobile 13.0.1.7.0

TDu committed:
[13.0] shopfloor_mobile: Add setting page improvements

Move the log out button to the main setting page.
On the profile and language setting buttons, add the
active value.

TDu committed:
[13.0] Add shopfloor_workstation

Shopfloor workstation are useful for a quick configuration of the shopfloor
mobile application and its connected user.

To synchronize a connected user to a workstation setting, go to
Settings - Workstation in the mobile application and scan a workstation barcode.

Commit:
shopfloor_mobile: Improve wording on "zero check"

The zero check screen shows 2 buttons:

* Confirm stock = 0
* Confirm stock not empty

At this point, the normal situation would be that the stock is zero.
So "Confirm stock not empty" is confusing, as we do not confirm the
expected value of zero.

simahawk committed:
shopfloor_mobile: bump 13.0.1.8.0

Commit:
shopfloor_mobile: bump 13.0.1.8.1

TDu committed:
shopfloor_mobile: settings show workstation name

This PR continue on the previous  ones. By keeping the name of the last
workstation scanned. And displaying it on the settings button, like for
profiles.

simahawk committed:
shopfloor_mobile: do not show edit carrier if no carrier set

simahawk committed:
shopfloor_mobile: fix call to update w/ new rest api

TDu committed:
shopfloor: Improve package and product details

Adds on the package details:

* the actual stock location of the package
* the barcode of the product
* the weight unit (kg)

And also add the barcode on the product details.

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (4 of 4 strings)

Translation: wms-13.0/wms-13.0-shopfloor_mobile
Translate-URL: https://translation.odoo-community.org/projects/wms-13-0/wms-13-0-shopfloor_mobile/es_AR/

OCA-git-bot committed:
shopfloor_mobile 13.0.1.9.0

simahawk committed:
shopfloor_mobile: fix menu

OCA-git-bot committed:
shopfloor_mobile 13.0.1.9.1

simahawk committed:
shopfloor_mobile: bump 13.0.1.10.0

simahawk committed:
shopfloor_mobile: allow custom icon for card title

simahawk committed:
shopfloor_mobile: checkout display picking carrier

OCA-git-bot committed:
shopfloor_mobile 13.0.1.11.0

simahawk committed:
shopfloor_mobile: depend on shopfloor_base

simahawk committed:
shopfloor_mobile: fix relative import paths

simahawk committed:
shopfloor_mobile: fix assets path

simahawk committed:
shopfloor_mobile: main.css split from base

simahawk committed:
shopfloor_mobile_base: settings pages from registry

simahawk committed:
Add shopfloor_workstation_mobile

OCA-git-bot committed:
shopfloor_mobile 13.0.1.12.0

simahawk committed:
shopfloor: split wms utils

simahawk committed:
sf_mobile_base: move wms components to sf_mobile

simahawk committed:
shopfloor_mobile: un-delete sf_m/.../misc.js

During the migration from v13 to v14,
shopfloor_mobile/static/wms/src/components/misc.js
has been wrongly deleted. (commit a2d0c12a2b777898a86)

This file was first in sf_mobile
then moved in sf_mobile_base (ee7374e86ef9bd960a3c499)
then added back in sf_mobile (3df3a176569bb380f4cef53)

This commit is a cherry-pick of 3df3a176569 from v13
sf_mobile_base: move wms components to sf_mobile

simahawk committed:
sf_mobile: override menu item templates

simahawk committed:
sf_mobile: wms_utils fix parseInt usage

simahawk committed:
sf_mobile: update ROADMAP

simahawk committed:
sf_mobile: improve pkg qty picker display control

* use: options.non_zero_only to hide pakaging w/ no qty
* use: options.pkg_name_key to control which field is used for pkg name

simahawk committed:
sf_mobile: checkout display qty using qty widget

simahawk committed:
sf_mobile: checkout fix package line select handling

simahawk committed:
sf_mobile: keep manual select box always on top

simahawk committed:
shopfloor_mobile: fix usage of parseInt

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update shopfloor_mobile.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-13.0/wms-13.0-shopfloor_mobile
Translate-URL: https://translation.odoo-community.org/projects/wms-13-0/wms-13-0-shopfloor_mobile/

Commit:
shopfloor: checkout, select the delivery packaging

Add a new state/screen when the user click on the "New package" button
to list available delivery packaging ('select_delivery_packaging' state)
for the current transfer and select one to create the package (existing
'new_package' method which now accepts a packaging as parameter).

OCA-git-bot committed:
shopfloor_mobile 13.0.2.0.1

OCA-git-bot committed:
shopfloor_mobile 13.0.2.1.0

simahawk committed:
sf_mobile: fix single pack transf confirm location change

Ensure that if we scan again the location
it gets confirmed without asking anymore.

OCA-git-bot committed:
shopfloor_mobile 13.0.2.2.0

OCA-git-bot committed:
shopfloor_mobile 13.0.2.2.1

simahawk committed:
sf_mobile: fix regression on picking detail widgets

simahawk committed:
sf_mobile: checkout block 'new package' if no carrier set

simahawk committed:
shopfloor: unify msg for packed goods

OCA-git-bot committed:
shopfloor_mobile 13.0.2.2.2

lmignon committed:
[FIX] shopfloor_mobile: Fix demo mode

OCA-git-bot committed:
shopfloor_mobile 13.0.2.2.3

simahawk committed:
sf_mobile: fix detail package show reserved qty

Use pkg qty instead of the whole qty reserved for the product!

hparfr committed:
shopfloor_mobile: remove files

From ee7374e86ef9bd960a3c49faa2826f280a6410b9
`git mv` files were not removed correctly

OCA-git-bot committed:
shopfloor_mobile 13.0.2.3.0

OCA-git-bot committed:
shopfloor_mobile 13.0.2.3.1

TDu committed:
[130][IMP] shopfloor mobile carrier display

In the backend add the carrier computed from the module
`stock_picking_delivery_link` into the picking data send.

In the frontend: checkout, set the color of the carrier card in green

* Add the carrier on the cluster picking last screen.
* On the checkout and the cluster picking improve the carrier display by using the one added in the backend if needed.
* Set the carrier card in green.

TDu committed:
Add easy pack change in zone picking (shopfloor)

On the zone picking scenario it is possible to select a move line
by scanning the pack assigned to the move line.

But for some situation (block storage) where a location contains
multiple pack of the same type it may be difficult to scan the exact
pack, and much easier to scan an identical one in front of the user.

This change allows for this so when scanning a pack that can replace a
pack in the existing move line, the user is offered the option to do so,
by scanning the same pack a 2nd time.

OCA-git-bot committed:
shopfloor_mobile 13.0.2.4.0

simahawk committed:
sf_mobile: fix demo checkout data

simahawk committed:
sf_mobile: qty picker display always pkg code

simahawk committed:
sf_mobile: unify pkg display usage

OCA-git-bot committed:
shopfloor_mobile 13.0.2.4.1

mmequignon committed:
shopfloor_mobile: Add hooks in picking detail scenarios

hparfr committed:
[IMP] shopfloor_mobile: black, isort, prettier

hparfr committed:
[MIG] shopfloor_mobile: Migration to 14.0

hparfr committed:
[MIG] shopfloor_mobile: eslint

hparfr committed:
shopfloor_mobile: remove dead code

oca-travis committed:
[UPD] Update shopfloor_mobile.pot

OCA-git-bot committed:
[UPD] README.rst

hparfr committed:
shopfloor_mobile: linting

OCA-git-bot committed:
shopfloor_mobile 14.0.1.0.1

simahawk committed:
shopfloor_mobile: 14.0.1.1.0

Add migration step for auth split

OCA-git-bot committed:
shopfloor_mobile 14.0.1.0.2

lmignon committed:
[FIX] shopfloor_mobile: Display location as TODO as long as the article is not scanned

marcelsavegnago committed:
Added translation using Weblate (Portuguese (Brazil))

jbaudoux committed:
shopfloor: refactor the packaging quantity picker

Take less space on screen. Use a grid system to better organize info.
Allow to edit total quantity.

OCA-git-bot committed:
shopfloor_mobile 14.0.1.2.0

jbaudoux committed:
[FIX] shopfloor: fix computation of contained packagings

lmignon committed:
shopfloor_mobile: add missing translations

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
shopfloor_mobile 14.0.1.3.0

OCA-git-bot committed:
shopfloor_mobile 14.0.1.3.1

OCA-git-bot committed:
shopfloor_mobile 14.0.1.4.0

Commit:
[FIX] shopfloor_mobile: fix translations

OCA-git-bot committed:
shopfloor_mobile 14.0.1.4.1

mmequignon committed:
shopfloor, checkout: Allow to scan products

OCA-git-bot committed:
shopfloor_mobile 14.0.1.5.0

TDu committed:
Add sublocation scan on shopfloor delivery scenario

When scanning a product barcode or a lot the scenario will search
for all related move lines and pick the first one by default.

For a shopfloor menu configuration with locations containing sub locations
this is not ideal. Because a user working at a sub location only wants
move lines related to that sub location to be selected.

To help with this situation, a user can first scan his working location
which is kept as long as no other location is scanned.
The sub location set can also be reset by the click of a button.

Also when the search for a move line to work on returns multiple move lines
related to different locations and no location has been preliminary scanned.
The user will be prompted to scann a location first.

JuMiSanAr committed:
sf_mobile: fix qty picker reactivity

When using packaging-qty-picker in display mode,
the qty_todo wasn't updated when scanning a new barcode
as it was only calculated when mounting the component.
An updated hook is now taking care of that.

JuMiSanAr committed:
sf_mobile: demo checkout add scan_package_action endpoint

OCA-git-bot committed:
shopfloor_mobile 14.0.1.5.1

JuMiSanAr committed:
shopfloor_mobile: single_pack_transfer frontend display pack

OCA-git-bot committed:
shopfloor_mobile 14.0.1.6.0

TDu committed:
shopfloor_mobile: allow extenstion of states on checkout

This change allows for other modules to extend the state of the checkout
scenario.
See the `shopfloor_checkout_package_measurement_mobile` for a use of
this change.

TDu committed:
shopfloor: fix zone picking label

TDu committed:
shopfloor: zone picking add unloading on first screen

The user may end up in the `scan_location` screen with some products to
unload. It can happen when clicking on the back button but also when
there is no lines left to process in the current zone.

This change adds an Unload At Destination button on the `scan_location`
screen which will be display only when the user has some product to
unload.

OCA-git-bot committed:
shopfloor_mobile 14.0.1.6.1

OCA-git-bot committed:
shopfloor_mobile 14.0.1.7.0

TDu committed:
shopfloor_mobile: fix zone picking unloading on first screen

OCA-git-bot committed:
shopfloor_mobile 14.0.1.7.1

TDu committed:
shopfloor: imp zp scan source product

Ff multiple move lines are found on several locations, raise an error
and filter the list with the matched moves.

If multiple move lines are found for different lots, raise an error
and filter the list with the matched moves.

OCA-git-bot committed:
shopfloor_mobile 14.0.1.8.0

OCA-git-bot committed:
shopfloor_mobile 14.0.1.9.0

TDu committed:
shopfloor: checkout add no prefill qty

Co-authored-by: Juan Miguel Sánchez Arce <juan.sanchez@camptocamp.com>
Co-authored-by: Matthieu Méquignon <matthieu.mequignon@camptocamp.com>
Co-authored-by: Jacques-Etienne Baudoux <je@bcim.be>

TDu committed:
shopfloor: zone picking add no prefill qty

Co-authored-by: Juan Miguel Sánchez Arce <juan.sanchez@camptocamp.com>
Co-authored-by: Matthieu Méquignon <matthieu.mequignon@camptocamp.com>
Co-authored-by: Jacques-Etienne Baudoux <je@bcim.be>

jbaudoux committed:
shopfloor_mobile: Fix qty widget - UOM at last

When there is a product packaging with qty 1, the product uom was not
displayed as the last packaging decomposition.
This change ensures the product uom is always displayed after the list
of product packagings.

jbaudoux committed:
shopfloor_mobile: Improve qty widget in readonly

Co-authored-by: Juan Miguel Sánchez Arce <juan.sanchez@camptocamp.com>

Show packagings if relevant

jbaudoux committed:
shopfloor_mobile: Qty picker initial quantity

Co-authored-by: Juan Miguel Sánchez Arce <juan.sanchez@camptocamp.com>

Allow to initialise the quantity done in the quantity picker.
Always show the quantity todo next to the done quantity.

jbaudoux committed:
shopfloor_mobile: selected move line color

Adapt the color of the move line based on the quantity done vs todo.
Use different colors if less, equal or greater.

JuMiSanAr committed:
shopfloor_mobile: checkout packaging list hide type

OCA-git-bot committed:
shopfloor_mobile 14.0.1.9.1

TDu committed:
shopfloor_mobile: checkout fix new package button

This button should not only be available when there is a carrier.

simahawk committed:
shopfloor: checkout improve lines hooks

* allow to turn off group by location
* ease override for lines manipulation

JuMiSanAr committed:
shopfloor_mobile: make picking_summary fields customizable

JuMiSanAr committed:
shopfloor_mobile: add group_lines_by_product in wms_utils

JuMiSanAr committed:
shopfloor_mobile: adapt frontend to stock_picking_progress

OCA-git-bot committed:
shopfloor_mobile 14.0.1.9.2

JuMiSanAr committed:
shopfloor_mobile: location_content_transfer get work

TDu committed:
shopfloor_mobile: imp zone picking screen title

If there is a current picking being worked on. Add it to the top
of the screen. And remove redondant information.

OCA-git-bot committed:
shopfloor_mobile 14.0.1.10.0

OCA-git-bot committed:
shopfloor_mobile 14.0.1.11.0

OCA-git-bot committed:
shopfloor_mobile 14.0.1.12.0

TDu committed:
shopfloor: fix zone picking scenario

Doing a scan of a location to restrict the search for the next scan on a
product, was not working.

OCA-git-bot committed:
shopfloor_mobile 14.0.1.13.0

OCA-git-bot committed:
shopfloor_mobile 14.0.1.14.0

OCA-git-bot committed:
shopfloor_mobile 14.0.1.15.0

simahawk committed:
shopfloor: cluster_picking full bin action opt-out

Add menu flag to disable the full bin action.
This way the operator is forced to finish the picking
using additional bins when full.

simahawk committed:
shopfloor: checkout allow pack all lines at once

On the select_line screen,
you can now skip the selection and scan a delivery package.

When this happens, the system asks for confirmation.

If you scan again the delivery package,
all lines will be selected and packed at the same time in the same package.

simahawk committed:
shopfloor_mobile: bump 14.0.1.16.0

TDu committed:
shopfloor: Add first scan location option on zp

Add a new scenario option `First scan location or pack`.
And implement it on the zone picking scenario.

OCA-git-bot committed:
shopfloor_mobile 14.0.1.17.0

OCA-git-bot committed:
shopfloor_mobile 14.0.1.18.0

OCA-git-bot committed:
shopfloor_mobile 14.0.1.19.0

simahawk committed:
shopfloor: checkout add show_oneline_package_content option

On the menu item you can decide to enable the option
'show_oneline_package_content'.
Once enabled, on the select line screen,
if the package contains only one line
the lot and the product name will be shown.

The use case it tries to solve:
you have multiple packs in front of you
and you want to be able to scan any of the packs
using the lot or the product.

OCA-git-bot committed:
shopfloor_mobile 14.0.1.20.0

OCA-git-bot committed:
shopfloor_mobile 14.0.1.21.0

simahawk committed:
shopfloor_mobile: avoid re-render searchbar on updates

The searchbar is always rendered optionally based on the state.
However, is pointless and slower to re-render it every time.
Just hide it instead.
This way the object stays in the DOM and any event bound to it will stay as well.

OCA-git-bot committed:
shopfloor_mobile 14.0.1.21.1

JuMiSanAr committed:
shopfloor_mobile: improve qty picker calculation

JuMiSanAr committed:
shopfloor_mobile: zone_picking always refresh list of lines

OCA-git-bot committed:
shopfloor_mobile 14.0.1.21.2

JuMiSanAr committed:
shopfloor_mobile: trigger manual_select event in checkout

OCA-git-bot committed:
shopfloor_mobile 14.0.1.21.3

jbaudoux committed:
shopfoor_mobile: fix checkout line selection

Fix get_wrapper_klass is not defined on checkout line selection
Fixes 52cce73848a80489f8953359da51f3be78ddc77d

JuMiSanAr committed:
shopfloor: display location operations progress in delivery scenario

JuMiSanAr committed:
shopfloor_mobile: add picking select detail todo

OCA-git-bot committed:
shopfloor_mobile 14.0.1.21.4

JuMiSanAr committed:
shopfloor: fix cluster_picking change_lot not returning qty done

OCA-git-bot committed:
shopfloor_mobile 14.0.1.21.5

TDu committed:
shopfloor: imp zp scan location or pack first

With the option scan_location_or_pack_first on.
When scanning a location, only the lines without a source package
are offered to be processed.
If none are found, ask for scanning a package.

TDu committed:
shopfloor: cp add option first scan location or pack

JuMiSanAr committed:
sf_single_pack_transfer: display all products in pack

JuMiSanAr committed:
shopfloor zone_picking: block move selection

OCA-git-bot committed:
shopfloor_mobile 14.0.1.22.0

OCA-git-bot committed:
shopfloor_mobile 14.0.1.22.1

OCA-git-bot committed:
shopfloor_mobile 14.0.1.23.0

OCA-git-bot committed:
shopfloor_mobile 14.0.1.24.0

JuMiSanAr committed:
shopfloor: checkout fix highlights bug

OCA-git-bot committed:
shopfloor_mobile 14.0.1.25.0

OCA-git-bot committed:
shopfloor_mobile 14.0.1.26.0

OCA-git-bot committed:
shopfloor_mobile 14.0.1.27.0

OCA-git-bot committed:
shopfloor_mobile 14.0.1.27.1

JuMiSanAr committed:
sf_single_pack_transfer: display weight in package info card

TDu committed:
sh_mobile: checkout add option barcode scan restritcion

simahawk committed:
shopfloor_mobile*: demo case now requires menu id

OCA-git-bot committed:
shopfloor_mobile 14.0.1.28.0

OCA-git-bot committed:
shopfloor_mobile 14.0.1.29.0

OCA-git-bot committed:
shopfloor_mobile 14.0.1.30.0

sbejaoui committed:
[16.0][MIG] - shopfloor_mobile

sbejaoui committed:
[FIX] shopfloor_mobile: bin card always step_todo as it's the last step

sbejaoui committed:
[FIX] bin card always step_todo as it's the last step

sbejaoui committed:
[IMP] shopfloor_mobile: allow searchbar input type override

sbejaoui committed:
[FIX] shopfloor_mobile: fix clustrer picking unload_scan_pack package id

nilshamerlinck commented on stock_release_channel/models/res_company.py:
```diff
@@ -0,0 +1,14 @@
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    recompute_release_channel_after_released = fields.Boolean(
```
```suggestion
    recompute_channel_on_pickings_at_release = fields.Boolean(
```

nilshamerlinck commented on stock_release_channel/models/res_company.py:
```diff
@@ -0,0 +1,14 @@
+    _inherit = "res.company"
+
+    recompute_release_channel_after_released = fields.Boolean(
+        string="Recompute Release Channel After Released",
```
```suggestion
        string="Recompute Channel On Pickings At Release",
```

nilshamerlinck commented on stock_release_channel/models/res_company.py:
```diff
@@ -0,0 +1,14 @@
+
+    recompute_release_channel_after_released = fields.Boolean(
+        string="Recompute Release Channel After Released",
+        help="Will automatically recompute release channel on the picking after released",
```
```suggestion
```

nilshamerlinck replied:
string is clear enough imho

nilshamerlinck commented on stock_release_channel/models/stock_move.py:
```diff
@@ -16,7 +16,11 @@ def release_available_to_promise(self):
         # they still exist.
         moves = self.exists()
-        moves.picking_id.assign_release_channel()
+        # Base on the flag that was configued on the Company,
+        # we decide to recompute the release channel after released picking
```
```suggestion
```

nilshamerlinck replied:
code is clear enough, no need for comment

nilshamerlinck commented on stock_release_channel/models/stock_move.py:
```diff
@@ -16,7 +16,11 @@ def release_available_to_promise(self):
+        # Base on the flag that was configued on the Company,
+        # we decide to recompute the release channel after released picking
+        company = self.env.company
+        if company.recompute_release_channel_after_released:
```
```suggestion
        if company.recompute_channel_on_pickings_at_release:
```

nilshamerlinck reviewed (changes requested):
pre-commit?

nilshamerlinck commented on stock_release_channel/models/res_company.py:
```diff
@@ -0,0 +1,13 @@
+    _inherit = "res.company"
+
+    recompute_channel_on_pickings_at_release = fields.Boolean(
+        help="When releasing a transfer, recompute channel on pickings",
```
I'm confused, you favor help over string?

QuocDuong1306 replied:
For pre-commit, nils
```************* Module stock_release_channel.models.res_company
stock_release_channel/models/res_company.py:10: [W8113(attribute-string-redundant), ResCompany] The attribute string is redundant. String parameter equal to name of variable

Check for Odoo modules using pylint......................................Failed
- hook id: pylint_odoo
- exit code: 4

************* Module stock_release_channel.models.res_company
stock_release_channel/models/res_company.py:10: [W8113(attribute-string-redundant), ResCompany] The attribute string is redundant. String parameter equal to name of variable

QuocDuong1306 replied:
On xml with `group` tag, it will render the string equal to the name of variable

nilshamerlinck commented on stock_release_channel/models/res_company.py:
```diff
@@ -0,0 +1,13 @@
+    _inherit = "res.company"
+
+    recompute_channel_on_pickings_at_release = fields.Boolean(
+        help="When releasing a transfer, recompute channel on pickings",
```
a transfer is a picking, so we can simplify this sentence:
> When releasing a transfer, recompute channel

lmignon committed:
[FIX] shopfloor_mobile: Fix pre-commit

OCA-git-bot committed:
Merge PR #698 into 16.0

Signed-off-by rousseldenis

Commit:
[UPD] Update shopfloor_mobile.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-16.0/wms-16.0-shopfloor_mobile
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-shopfloor_mobile/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (2 of 2 strings)

Translation: wms-16.0/wms-16.0-shopfloor_mobile_base
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-shopfloor_mobile_base/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (5 of 5 strings)

Translation: wms-16.0/wms-16.0-shopfloor_mobile_base_auth_api_key
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-shopfloor_mobile_base_auth_api_key/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: wms-16.0/wms-16.0-shopfloor_rest_log
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-shopfloor_rest_log/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (9 of 9 strings)

Translation: wms-16.0/wms-16.0-shopfloor_batch_automatic_creation
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-shopfloor_batch_automatic_creation/it/

mymage committed:
Added translation using Weblate (Italian)

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (5 of 5 strings)

Translation: wms-16.0/wms-16.0-shopfloor_mobile_base_auth_api_key
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-shopfloor_mobile_base_auth_api_key/pt_BR/

Commit:
[IMP] stock_release_channel: Add a company setting to enable/disable recomputation of channel on release
